api/onix.py

```diff
@@ -0,0 +1,281 @@
+        'B15': Contributor.EDITOR_ROLE, # 'Editorial coordination by'
+        'B16': Contributor.EDITOR_ROLE, # Managing editor
+        'B17': Contributor.EDITOR_ROLE, # Founding editor of a serial publication
+        'B18': Contributor.EDITOR_ROLE, # 'Prepared for publication by'
+        'B19': Contributor.EDITOR_ROLE, # Associate editor
+        'B20': Contributor.EDITOR_ROLE, # Consultant editor
+        'B21': Contributor.EDITOR_ROLE, # General editor
+        'B22': Contributor.UNKNOWN_ROLE, # 'Dramatized by'
+        'B23': Contributor.EDITOR_ROLE, # 'General rapporteur'
+        'B24': Contributor.EDITOR_ROLE, # Literary editor
+        'B25': Contributor.COMPOSER_ROLE, # 'Arranged by (music)'
+        'B26': Contributor.EDITOR_ROLE, # Technical editor
+        'B27': Contributor.UNKNOWN_ROLE, # Thesis advisor
+        'B28': Contributor.UNKNOWN_ROLE, # Thesis examiner
+        'B29': Contributor.EDITOR_ROLE, # Scientific editor
+        'B30': Contributor.UNKNOWN_ROLE, # Historical advisor
+        'B31': Contributor.UNKNOWN_ROLE, # Editor of the first edition who is not an editor of the current edition
+        'B99': Contributor.EDITOR_ROLE, # Other type of adaptation or editing
+        'C01': Contributor.UNKNOWN_ROLE, # 'Compiled by'
+        'C02': Contributor.UNKNOWN_ROLE, # 'Selected by'
+        'C03': Contributor.UNKNOWN_ROLE, # 'Non-text material selected by'
+        'C04': Contributor.UNKNOWN_ROLE, # 'Curated by'
+        'C99': Contributor.UNKNOWN_ROLE, # Other type of compilation
+        'D01': Contributor.PRODUCER_ROLE,
+        'D02': Contributor.DIRECTOR_ROLE,
+        'D03': Contributor.MUSICIAN_ROLE, # Conductor
+        'D04': Contributor.UNKNOWN_ROLE, # Choreographer
+        'D05': Contributor.DIRECTOR_ROLE, # Other type of direction
+        'E01': Contributor.ACTOR_ROLE,
+        'E02': Contributor.PERFORMER_ROLE, # Dancer
+        'E03': Contributor.NARRATOR_ROLE, # 'Narrator'
+        'E04': Contributor.UNKNOWN_ROLE, # Commentator
+        'E05': Contributor.PERFORMER_ROLE, # Vocal soloist
+        'E06': Contributor.PERFORMER_ROLE, # Instrumental soloist
+        'E07': Contributor.NARRATOR_ROLE, # Reader of recorded text, as in an audiobook
+        'E08': Contributor.PERFORMER_ROLE, # Name of a musical group in a performing role
+        'E09': Contributor.PERFORMER_ROLE, # Speaker
+        'E10': Contributor.UNKNOWN_ROLE, # Presenter
+        'E99': Contributor.PERFORMER_ROLE, # Other type of performer
+        'F01': Contributor.PHOTOGRAPHER_ROLE, # 'Filmed/photographed by'
+        'F02': Contributor.EDITOR_ROLE, # 'Editor (film or video)'
+        'F99': Contributor.UNKNOWN_ROLE, # Other type of recording
+        'Z01': Contributor.UNKNOWN_ROLE, # 'Assisted by'
+        'Z02': Contributor.UNKNOWN_ROLE, # 'Honored/dedicated to'
+        'Z99': Contributor.UNKNOWN_ROLE, # Other creative responsibility
+    }
+    
+    @classmethod
+    def parse(cls, file, data_source_name):
```

Comment:
What about making this a subclass of core/xmlparser.py:XMLParser? I think many chunks of this code could each be simplified by using _xpath and _xpath1. It would also handle namespaces, though judging from your sample, ONIX files aren't super concerned with namespaces.

Reply:
Yeah, I don't think I could do much with namespaces but I could probably use some of the parser methods more effectively.